### ISSUE TITLE
Cleanup istio labels in Kiali config

### ISSUE THREAD

**Issue by nrfox** [kiali/kiali#7722]
### What do you want to improve?

Most, if not all, of the `IstioLabels` configuration in Kiali are actually constants that do not change: https://github.com/kiali/kiali/blob/c39657b25440cabe97a205c754a50d8fcf02d031/config/config.go#L338-L350

These can be removed from the user facing Kiali configuration and turned into internal only constants.

**Comment by jshaughn** [comment]
@codeshekh Sorry, I was out for the US holiday.  Yes, thanks!  Please work with @hhovsepy.

**Comment by smc181002** [comment]
Hey @jshaughn, Is this issue still open? I’d like to contribute if it is.

Based on the Kiali docs, `AppLabelName` and `VersionLabelName` still need to remain configurable to support Istio’s custom label names.

My current plan is to define the default values as constants and then make these fields internal by changing their struct tags to <code>&#x60;yaml:"-" json:"-"&#x60;</code>. The `NewConfig()` function will be using these Const values.

This would keep the values out of user-facing YAML configuration while minimising changes across the code base, since most existing references to these fields could remain intact.

If you think it’s better to replace the direct field references with constants everywhere, I’m happy to take that approach instead.

**Comment by jshaughn** [comment]
It looks open, @nrfox , what do you think?

**Comment by nrfox** [comment]
> If you think it’s better to replace the direct field references with constants everywhere

I would do this.

> Based on the Kiali docs, AppLabelName and VersionLabelName still need to remain configurable to support Istio’s custom label names.

That's probably true. @jshaughn I don't remember doesn't Kiali use either or both `app` and `app.kubernetes.io/name` as labels?

**Comment by smc181002** [comment]
> I don't remember doesn't Kiali use either or both `app` and `app.kubernetes.io/name` as labels?

@nrfox Right. This is the document reference: 
* [Kiali - Istio Environment](https://kiali.io/docs/configuration/istio/#labels-and-resource-names)
* [Istio - Pod Requirements -  pod labels](https://istio.io/latest/docs/ops/deployment/application-requirements/#pod-requirements:~:text=Pod%20labels%3A%20We,%2C%20or%20version)

Istio allows the following label names
* Application name: `service.istio.io/canonical-name`, `app.kubernetes.io/name`, or `app`.
* Application version: `service.istio.io/canonical-revision`, `app.kubernetes.io/version`, or `version`.

> I would do this.

Great. I'll go ahead and start working on the changes if you agree.

**Comment by smc181002** [comment]
Gentle ping @jshaughn @nrfox 

Could I start working on the issue?

I plan to update all the references other than `AppLabelName` and `VersionLabelName` to const values in `config/config.go`.
I will try to reuse the existing consts in the `config.go`
Then I will update all the references of these attributes with the constants.

Any comments/changes from the proposed approach?

As for tests, I'm not sure if we require any additional tests as this is a non-functional change. Please suggest if you think otherwise.

**Comment by jshaughn** [comment]
@smc181002 , yes, go ahead. And correct, stay away from appLabelName and versionLabelName. Those have special handling to allow for a fixed or mixed set of label names. I'm unsure about additional tests, first we can see if it passes the existing CI, and if so we can investigate any test gaps.
